src/Surface.zig

```diff
@@ -4119,6 +4119,17 @@ pub fn performBindingAction(self: *Surface, action: input.Binding.Action) !bool
             }, .unlocked);
         },
 
+        .scroll_to_selection => {
+            const sel = self.io.terminal.screen.selection orelse return false;
+            const pin = sel.start();
+            // TODO: If the selection is outside the active area and smaller
+            // than the viewport, we should move the pin backwards a little so
+            // that the selection is vertically centered.
```

Comment:
The TODO I think is unnecessary, we can consider this a future improvement if we want.